### ISSUE TITLE
Why is the line written twice 

### ISSUE THREAD

**Issue by HeSitates** [nunit/docs#104]
I'm trying to create a demo project for my colleagues, but I'll notice something unexpected.

```
[SetUpFixture]
public class SetUpFixtureNotInNameSpace
{
    [OneTimeSetUp]
    public void Init()
    {
        WriteToContext();
    }

    [OneTimeTearDown]
    public void CleanUp()
    {
        WriteToContext();
    }

    private void WriteToContext([CallerMemberName] string text = "")
    {
        TestContext.WriteLine(string.Format( "SetUpFixture Not In NameSpace - {0}: {1:dd-MM-yyyy HH:mm:ss.fff}",  text, DateTime.Now));
    }
}
```

I thought it should only once to the output, but it does it twice:

> => SetUpFixtureNotInNameSpace
> SetUpFixture Not In NameSpace - Init: 17-08-2016 21:30:05.561
> SetUpFixture Not In NameSpace - CleanUp: 17-08-2016 21:30:05.640
> SetUpFixture Not In NameSpace - Init: 17-08-2016 21:30:05.561
> SetUpFixture Not In NameSpace - CleanUp: 17-08-2016 21:30:05.640

Strangely the time is the same and the debugger only stops once....

Is the error between keyboard and chair? :)

**Comment by CharliePoole** [comment]
This is a crazy guess, but is your TrstFixture inheriting from your SetUpFixture?

**Comment by HeSitates** [comment]
Not a crazy guess, but it is not.

I've not implemented any inheritance in my demo project yet.

**Comment by CharliePoole** [comment]
What happens if you eliminate the CallerMemberNameAttribute and specify the text in each call?

**Comment by HeSitates** [comment]
I've changed my test into (not a test best-practice :), but it proves this situation):
```
public class SetUpFixtureNotInNameSpace
{
    private double Counter { get; set; }

    [OneTimeSetUp]
    public void Init()
    {
        WriteToContextLog( Counter++, "Init" );
    }

    [OneTimeTearDown]
    public void CleanUp()
    {
        WriteToContextLog( Counter++, "CleanUp" );
    }

    private static void WriteToContextLog( double counter, string text = "")
    {
        TestContext.WriteLine( string.Format( "{0}: {1} - {2:dd-MM-yyyy HH:mm:ss.fff}", text, counter, DateTime.Now ) );
    }
}
```

Now the output is:

> => SetUpFixtureNotInNameSpace
> Init: 0 - 18-08-2016 10:21:00.485
> CleanUp: 1 - 18-08-2016 10:21:00.596
> Init: 0 - 18-08-2016 10:21:00.485
> CleanUp: 1 - 18-08-2016 10:21:00.596

So my idea is that it is runned only once, but printed twice.

**Comment by jplebre** [comment]
Sorry to complicate things, but I was trying to replicate your issue, and I had another issue:

![image](https://cloud.githubusercontent.com/assets/5503088/17768081/66485e1c-652a-11e6-917d-75e4e3950830.png)

this is with .NET4.6.1, latest stable release n-unit

Which platform are you using?

**Comment by HeSitates** [comment]
I'm using:

> NUnit Console Runner 3.4.1
> Copyright (C) 2016 Charlie Poole
> 
> Runtime Environment
>    OS Version: Microsoft Windows NT 10.0.10586.0
>   CLR Version: 4.0.30319.42000

Also using Resharper (Ultimate 2016.1) but there the lines from the contextwriter is not shown at all for the setupfixtures.

**Comment by jplebre** [comment]
Same setup using dotnet core and 3.4.0-beta-1 returns a slightly diff message:
```
------ Run test started ------
Running tests in 'C:\Users\jplebre\Desktop\NUnit.Tests1\ClassLibrary1\project.json' ["C:\Program Files\dotnet\dotnet.exe" test "C:\Users\jplebre\Desktop\NUnit.Tests1\ClassLibrary1\project.json" --output "C:\Users\jplebre\Desktop\NUnit.Tests1\ClassLibrary1\bin\Debug\netcoreapp1.0" --port 57901 --parentProcessId 532 --no-build]
No handler for message 'TestRunner.TestCompleted' when at state 'TestExecutionSentTestRunnerProcessStartInfo'
An exception occurred while invoking executor 'executor://dotnetunittestexecutor/v1': Unable to write data to the transport connection: An existing connection was forcibly closed by the remote host.
An existing connection was forcibly closed by the remote host
========== Run test finished: 0 run (0:00:00.7196596) ==========
```

I'm not sure what the issue is, just trying to replicate it, maybe it'll help find the issue

**Comment by SeanKilleen** [comment]
🧹 Cleanup -- I am closing this issue because it is 3+ years old and does not appear to have an actionable resolution for the NUnit docs project. 

If you feel this is closed in error, don't hesitate to ask me to reopen. Thanks!
